src/main/scala/PropertyChecker.scala

```diff
@@ -630,4 +664,285 @@ object PropertyChecker {
       }
     }
   }
+
+  def uriTemplateProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+          // Don't know how to place a URI object in JsonNode, keeping the text value as of now
+        case s: TextNode => (s, Array[String](), csvwPropertyType)
+        case _ => (new TextNode(""), Array[String](PropertyChecker.invalidValueWarning), csvwPropertyType)
+      }
+    }
+  }
+
+  def textDirectionProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      value match {
+        case s: TextNode if Array[String]("ltr", "rtl", "inherit").contains(s.asText()) => {
+          (value, Array[String](), PropertyType.Inherited)
+        }
+        case _ => (new TextNode(PropertyType.Inherited.toString), Array[String](PropertyChecker.invalidValueWarning), PropertyType.Inherited)
+      }
+    }
+  }
+
+  def naturalLanguageProperty(csvwPropertyType: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    (value, baseUrl, lang) => {
+      var warnings = Array[String]()
+      value match {
+        case s:TextNode => {
+          val returnObject = JsonNodeFactory.instance.objectNode()
+          val arrayNode = JsonNodeFactory.instance.arrayNode()
+          arrayNode.add(s.asText())
+          returnObject.set("lang", arrayNode)
+          (returnObject, Array[String](), csvwPropertyType)
+        }
+        case a:ArrayNode => {
+          var validTitles = Array[String]()
+          val arrayNodeIterator = a.elements()
+          while(arrayNodeIterator.hasNext()) {
```

Comment:
You can just use the `a.elements().asScala()` function to convert this into something Scala supports more natively.

Reply:
Tried using this method to see if that works, but now I have changed it to the way you suggested